robot/camera.py

```diff
@@ -17,7 +17,15 @@ class ResultList(List[Marker]):
     without checking that the list has any items.
     """
 
+    @overload
     def __getitem__(self, index: int) -> Marker:
+        ...
+
+    @overload  # noqa: F811 (deliberately method replacement)
+    def __getitem__(self, index: slice) -> List[Marker]:
+        ...
+
+    def __getitem__(self, index):  # noqa: F811 (deliberately method replacement)
```

Comment:
Are we able to define this as `Union[int, slice]` explicitly, or does the type checker work that out? Currently passing anything other than `int` or `slice` will be allowed at the type level and return `Any`, I think?

Reply:
I don't know whether the checker would allow `Union[int, slice]` if we changed the return type to `Any`. I suspect that it would then still error that it doesn't match the parent type.

While it looks like duplication, I think that what's here is the right thing to have.